### PR TITLE
maintenance/bitmart exchange unit tests

### PR DESCRIPTION
**Pull request by petioptrv**
**Before submitting this PR, please make sure**:

- [X] Your code builds clean without any errors or warnings
- [X] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
This PR adds unit tests for the Bitmart connector. It also refactors parts of the code where issues were discovered along the way.

**Tests performed by the developer**:
The entire PR

**Tips for QA testing**:
- Test that all orders are successfully cancelled when the run is stopped, as the function for this behaviour has been refactored.

### CODE REVIEW

**Review by rxlxrxsx — APPROVED** [review]
Tested and confirmed working by QA.

**Review by CrimsonJacket — CHANGES_REQUESTED** [review]
One minor comment. But other than that, unit tests LGTM.

File: hummingbot/connector/exchange/bitmart/bitmart_api_user_stream_data_source.py [hunk 1]
```diff
@@ -28,15 +28,25 @@ def logger(cls) -> HummingbotLogger:
...
+    def __init__(
+        self,
+        bitmart_auth: BitmartAuth,
+        throttler: Optional[AsyncThrottler] = None,
+        trading_pairs: Optional[List[str]] = None,
+    ):
         super().__init__()
         self._bitmart_auth: BitmartAuth = bitmart_auth
-        self._trading_pairs = trading_pairs
+        self._trading_pairs = trading_pairs or []
```
[the remark this answers is not in the document]
Comment (petioptrv) [reply hummingbot/connector/exchange/bitmart/bitmart_api_user_stream_data_source.py:44]: The problem is that the default value is then an object shared across instances, so technically default values should be immutable. Same goes for having empty dictionaries as default values, etc

[the remark this answers is not in the document]
Comment (aarmoa) [reply hummingbot/connector/exchange/bitmart/bitmart_api_user_stream_data_source.py:44]: I agree with @petioptrv on this one: we should avoid having mutable objects as parameters defaults

**Review by CrimsonJacket — APPROVED** [review]
Changes LGTM

**Pull request merged**